ee/errgroup/errgroup.go

```diff
@@ -0,0 +1,163 @@
+package errgroup
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"time"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sync/errgroup"
+)
+
+type LoggedErrgroup struct {
+	errgroup *errgroup.Group
+	cancel   context.CancelFunc
+	doneCtx  context.Context // nolint:containedctx
+	slogger  *slog.Logger
+}
+
+const (
+	maxShutdownGoroutineDuration = 3 * time.Second
+)
+
+func NewLoggedErrgroup(ctx context.Context, slogger *slog.Logger) *LoggedErrgroup {
+	ctx, cancel := context.WithCancel(ctx)
+	e, doneCtx := errgroup.WithContext(ctx)
+
+	return &LoggedErrgroup{
+		errgroup: e,
+		cancel:   cancel,
+		doneCtx:  doneCtx,
+		slogger:  slogger,
+	}
+}
+
+// StartGoroutine starts the given goroutine in the errgroup, ensuring that we log its start and exit.
+func (l *LoggedErrgroup) StartGoroutine(ctx context.Context, goroutineName string, goroutine func() error) {
+	l.errgroup.Go(func() (err error) {
+		slogger := l.slogger.With("goroutine_name", goroutineName)
+
+		// Catch any panicking goroutines and log them. We also want to make sure
+		// we return an error from this goroutine overall if it panics.
+		defer func() {
```

Comment:
do we want to add the same or similar defer func to catch panics to StartRepeatedGoroutine and AddShutdownGoroutine?

Reply:
They both call `StartGoroutine` so I think it should be covered -- I will add tests to confirm!

Reply:
ohh woops, duh, I see it now =) thx

Reply:
It does work -- however, I noticed that setting the error after recovering from the panic is not what we actually want to do during the shutdown goroutine. I will get that fixed up and then push both tests. 🙂

Reply:
Updated!